### ISSUE TITLE
RGB-OLED init and set number blocks

### ISSUE THREAD

**Issue by CarsonGracey** [parallaxinc/BlocklyProp#1166]
Currently in demo there is a issue with defined macros among ColorPal and RGB blocks. I can make a separate issue for each of the three problem blocks but they are all related. I did make the macros work but it was messy and this seems much simpler (I think).

These blocks

![capture3](https://user-images.githubusercontent.com/20099916/29638659-bf89d388-880c-11e7-9bec-a05a75556702.PNG)

Produce this code (I highlighted the parts that need to be changed to integers)

![capture2](https://user-images.githubusercontent.com/20099916/29639731-a304d7cc-8810-11e7-8474-3fc2f536238f.PNG)

With this error message because it wants to return something like "LED_COUNT1 1" instead of "LED_COUNT1" but that adds too many arguments in the function.

Required libraries: colorpal, simpletools, ws2812
Library compile order: None needed
single.c: In function 'main':
single.c:30:28: error: 'LED_COUNT' undeclared (first use in this function)
single.c:30:28: note: each undeclared identifier is reported only once for each function it appears in

---

One fix would be to remove both instances of # define (created by the RGB-Init block) and change all instances of LED_PIN1 and LED_COUNT1 to integers. 

![capture4](https://user-images.githubusercontent.com/20099916/29640071-fbd00862-8811-11e7-9a3d-ab34291580c3.PNG)

# Individual blocks

![capture](https://user-images.githubusercontent.com/20099916/29640299-259db15c-8813-11e7-9080-469a398e7682.PNG)

Delete lines 5 and 6

![capture1](https://user-images.githubusercontent.com/20099916/29640365-853456b6-8813-11e7-9784-6780aa9c5e14.PNG)

---

![capture](https://user-images.githubusercontent.com/20099916/29640547-45094d2a-8814-11e7-8483-ac4f59221d01.PNG)

Only the bottom one line 22 change to an int

![capture1](https://user-images.githubusercontent.com/20099916/29640591-729aa2ca-8814-11e7-9537-b574ecc13416.PNG)

---

![capture2](https://user-images.githubusercontent.com/20099916/29640628-9eea54f6-8814-11e7-9df0-49bd4be5d4ea.PNG)

Only the bottom one line 24 to an int

![capture3](https://user-images.githubusercontent.com/20099916/29640643-b83e7f4a-8814-11e7-92fa-2df5b31709c9.PNG)

---

![capture4](https://user-images.githubusercontent.com/20099916/29640669-de9c2eb2-8814-11e7-9b67-1b133eed2712.PNG)

Only the bottom one again both line 19

![capture5](https://user-images.githubusercontent.com/20099916/29640709-0393cf54-8815-11e7-91fe-845a5abc58a8.PNG)

**Comment by CarsonGracey** [comment]
These changes are better than what I had proposed, tested and verified.
